operator/pkg/component/component.go

```diff
@@ -653,7 +653,7 @@ func renderManifest(c IstioComponent, cf *CommonComponentFields) (string, error)
 		scope.Infof("Initial manifest with merged values:\n%s\n", my)
 	}
 	// Add the k8s resources from IstioOperatorSpec.
-	my, err = cf.Translator.OverlayK8sSettings(my, cf.InstallSpec, cf.componentName, cf.index)
+	my, err = cf.Translator.OverlayK8sSettings(my, cf.InstallSpec, cf.componentName, cf.resourceName, cf.index)
```

Comment:
where is the resource name populated? I can see we populate that when we create the gateway components, what about other components?

Reply:
if it's not populated we end up getting the default from the translate config file.